Sources/HummingbirdCore/Response/Response.swift

```diff
@@ -16,32 +16,34 @@ import HTTPTypes
 
 /// Holds all the required to generate a HTTP Response
 public struct Response: Sendable {
-    public var head: HTTPResponse
+    @inlinable
+    public var head: HTTPResponse {
+        get { HTTPResponse(status: status, headerFields: headers) }
+        set {
+            self.status = newValue.status
+            self.headers = newValue.headerFields
+        }
+    }
+
     public var body: ResponseBody {
         didSet {
             if let contentLength = body.contentLength {
-                self.head.headerFields[.contentLength] = String(describing: contentLength)
+                self.headers[.contentLength] = String(describing: contentLength)
             }
         }
     }
 
     public init(status: HTTPResponse.Status, headers: HTTPFields = .init(), body: ResponseBody = .init()) {
-        self.head = .init(status: status, headerFields: headers)
+        self.status = status
+        self.headers = headers
         self.body = body
-        if let contentLength = body.contentLength, headers[.contentLength] == nil {
-            self.head.headerFields[.contentLength] = String(describing: contentLength)
+        if let contentLength = body.contentLength, self.headers[.contentLength] == nil {
+            self.headers[.contentLength] = String(describing: contentLength)
         }
     }
 
-    public var status: HTTPResponse.Status {
-        get { self.head.status }
-        set { self.head.status = newValue }
-    }
-
-    public var headers: HTTPFields {
-        get { self.head.headerFields }
-        set { self.head.headerFields = newValue }
-    }
+    public var status: HTTPResponse.Status
+    public var headers: HTTPFields
```

Comment:
First of all, modifying `response.headers` was leading to a CoW. This is because `headers` was a getter/setter, thus copied the value from `head`. I've reversed this relationship